src/NuGet.Core/NuGet.CommandLine.XPlat/ReportRenderers/IReportRenderer.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.CommandLine.XPlat
+{
+    internal interface IReportRenderer
```

Comment:
Suggestion: Make the IReportRendered independent of exit code

```mermaid
classDiagram
IReportRenderer~T~ <|-- ConsoleWriter : implements
IReportRenderer~T~ <|-- JsonPackageRendered : implements
IReportRenderer~T~ : TextWriter writer
IReportRenderer~T~ : WriteOutput(reportContent)
```
I think exit code is independent of text output.

Reply:
I left the same feedback as @dominoFire in a separate comment.